tensorboard/plugins/profile/pod_viewer/topology_graph/topology-graph.ts

```diff
@@ -0,0 +1,594 @@
+  },
+  /**
+   * Returns the x-axis location for the xNode'th node of the xChip'th chip of
+   * the xHost'th host.
+   */
+  getNodeXLoc: function(xHost: number, xChip: number, xNode: number): number {
+    return this.getChipXLoc(xHost, xChip) + xNode * this._nodeGridWidth;
+  },
+  /**
+   * Returns the location for each host in the system.
+   */
+  createHostData: function(
+      hostXDim: number, hostYDim: number): Array<HostData> {
+    let hostData = [];
+    for (let i = 0; i < hostXDim; i++) {
+      for (let j = 0; j < hostYDim; j++) {
+        hostData.push({xdim: i, ydim: j});
+      }
+    }
+    return hostData;
+  },
+  /**
+   * Draw the labels on x-axis and y-axis.
+   */
+  drawLabels: function(svg: any, xdims: number[], ydims: number[]) {
+    let parent = this;
+
+    // Draw label on x axis.
+    svg.selectAll('.xLabel')
+       .data(xdims)
+       .enter()
+       .append('text')
+       .text((d) => d)
+       .attr('x',
+             (d, i) => parent.getChipXLoc(
+                 Math.floor(i / parent._hostXStride), i % parent._hostXStride))
+       .attr('y', 0)
+       .style('text-anchor', 'middle')
+       .attr('transform', 'translate(' + CHIP_GRID_SIZE / 2 + ', -6)')
+       .attr('class', 'axis');
+
+    // Draw label on y axis.
+    svg.selectAll('.yLabel')
+       .data(ydims)
+       .enter()
+       .append('text')
+       .text((d) => d)
+       .attr('x', 0)
+       .attr('y',
+             (d, i) => parent.getChipYLoc(
+                 Math.floor(i / HOST_Y_STRIDE), i % HOST_Y_STRIDE))
+       .style('text-anchor', 'middle')
+       .attr('transform', 'translate(-12,' + CHIP_GRID_SIZE / 2 + ')')
+       .attr('class', 'axis');
+  },
+  /**
+   * Draw the UI of host cards.
+   */
+  drawHostCards: function(svg, data, gridWidth: number, gridHeight: number) {
+    let cards = svg.selectAll('.xdim').data(data, (d) => d.xdim);
+    cards.enter()
+         .append('rect')
+         .attr('x', (d) => d.xdim * gridWidth)
+         .attr('y', (d) => d.ydim * gridHeight)
+         .attr('rx', 4 * gridWidth / gridHeight)
+         .attr('ry', 4)
+         .attr('class', 'hour bordered')
+         .attr('width', gridWidth - HOST_TO_HOST_MARGIN)
+         .attr('height', gridHeight - HOST_TO_HOST_MARGIN)
+         .attr('border', 1)
+         .style('fill', 'F0F0F0')
+         .style('stroke', 'black')
+         .style('stroke-width', 1)
+         .merge(cards)
+         .transition()
+         .duration(1000);
+    cards.exit().remove();
+  },
+  /**
+   * Draw the UI of node cards.
+   */
+  drawNodeCards: function(svg: any, data: Array<TopoData>, colorScale: any) {
+    let parent = this;
+    let cards = svg.selectAll('.xdim').data(data, (d) => d.xdim);
+    cards.enter()
+         .append('rect')
+         .attr('id', (d) => 'rid' + d.rid)
+         .attr('x', (d) => parent.getNodeXLoc(
+                             Math.floor(d.xdim / parent._hostXStride),
```

Comment:
Indentation and formatting is kinda off. When something looks busy like this, please consider using more complete arrow function.
 ```ts
.attr('x', (d) => {
   return parent.getNodeXLoc(
       Math.floor(d.xdim / parent._hostXStride),
       d.xdim % parent._hostXStride, d.nid);
})
```

Reply:
Done.